test.js

```diff
@@ -275,12 +318,25 @@ function processURL(url) {
    });
 }
 
+function structuredLogger(level, message, details = {}) {
+   console.log(JSON.stringify({ level, message, ...details, timestamp: new Date().toISOString() }));
+}
 
 function processNextOption(reset_mathjax) {
    if (!keyid) {
       if (all_diffs.length) console.log('ALL DIFFS', all_diffs);
+      //Added for special cases ---------------------------------------------------------------------
+      //Description: Display all special cases
+      if (all_special.length) console.log('ALL SPECIAL', all_special);
+      //----------------------------------------------------------------------Added for special cases
       console.log('No more data to process');
-      console.log('SUMMARY: match', nmatch, 'diff', ndiff, 'new', nnew);
+      console.log('SUMMARY: match', nmatch, 'diff', ndiff, 'new', nnew, 'special', nspecial); // added for special cases
+
+      // Description: If one file pair differs, the test fails
+      if (ndiff > 0) {
+         structuredLogger('ERROR', 'Not all files match', { diffCount: ndiff });
```

Comment:
Are you using 'ERROR' string to catch problem in the jsroot workflow?

Reply:
This is the error message, if there is a diff:

> SUMMARY: match 44 diff 1 new 0
> {"level":"ERROR","message":"Not all files match","diffCount":1,"timestamp":"2024-04-18T07:48:59.952Z"}
> Error: Process completed with exit code 1.

I think it's just for printing.